### DISCUSSION TITLE
openweather integration

### DISCUSSION THREAD

**Discussion by thommierother** [flobz/psa_car_controller#143] [Q&A] [answered]
Which value do I need to connect the controller to the openweather api?  I already have the api key, but what is the syntax used in the config file?

**Comment by flobz** [comment, marked as answer]
simply "weather_api": "xxxxxxxxxxxxxxxx"

**Reply by nuiler** [reply]
Where can I find the config file under home assistant?

**Reply by kurim** [reply]
You may need to create a backup, extract it and edit the files.
But you need to re-issue the login after restore :(
I would also think that it will be better to place them outside of the HA docker container in the /config or /data filesystem folder.

**Reply by dreimer1986** [reply]
THREAD NECRO: There is a config.ini and the other known configs, all full of things to edit and tinker with. Some containers like zigbee2mqtt chose to allow a folder inside /share for configs. Maybe this should be possible in the docker, too. Like optional linking of /data in docker to /share/psacc outside. If config is found there, use this one and not the internal. Or the parameters need a Web UI edit possibility. This would be a problem when you update though...

**Comment by Jordan87** [comment]
Like this?
![image](https://github.com/flobz/psa_car_controller/assets/10414918/05bd139f-8297-483a-b85e-cdb9f6555dee)
Or this should be in format like rest of the parameters?
`weather_api = xxxxxxxxxxxxxxxx`
**co2_api** is correct for CO2 api?

**Reply by grahamrjuk** [reply]
I suggest it goes in config.json, not config.ini. In my config.json, before adding the API key there was a blank:
`"weather_api": ""` to which I added my api key.

My problem now is that the weather data is retrieved, but it looks like the wrong component is extracted.
From an example api call I get:
```
{
    "lat": 33.44,
    "lon": -94.04,
    "timezone": "America/Chicago",
    "timezone_offset": -18000,
    "current": {
        "dt": 1711232179,
        "sunrise": 1711196105,
        "sunset": 1711240195,
        "temp": 18.95,
        "feels_like": 18.34,
        "pressure": 1017,
        "humidity": 55,
        "dew_point": 9.72,
        "uvi": 1.71,
        "clouds": 0,
        "visibility": 10000,
        "wind_speed": 3.13,
        "wind_deg": 39,
        "wind_gust": 5.81,
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01d"
            }
        ]
    }
}
```

but the code tries to extract
`temp = weather_res_json.get("current", weather_res_json.get("main"))["temp"]`
I read this to mean From `main` get `current` and in that look for `temp` - but I could be wrong.

From my call it just needs to extract `current` and look for `temp`

In my error log at an exception for the above line of code I see:
`TypeError: 'NoneType' object is not subscriptable`
I am assuming that is because `main` is not where it is currently expected - happy to be corrected if I am wrong.

**Comment by JauntyJosef** [comment]
I have the same error message. 

`2024-08-16 10:17:51,111 :: ERROR :: Unable to get temperature from openweathermap : Traceback (most recent call last): File "/usr/local/lib/python3.9/dist-packages/psa_car_controller/psacc/utils/utils.py", line 20, in get_temp temp = weather_res_json.get("current", weather_res_json.get("main"))["temp"] TypeError: 'NoneType' object is not subscriptable`

I put the Openweathermap API Key in the config.json file.
